packages/patternfly-4/react-table/src/components/Table/Table.tsx

```diff
@@ -0,0 +1,298 @@
+
+export interface IExtraRowData {
+  rowIndex?: number;
+  rowKey?: RowKeyType;
+}
+
+export interface IExtraColumnData {
+  columnIndex?: number,
+  column?: IColumn,
+  property?: string,
+}
+
+export interface IExtraData extends IExtraColumnData, IExtraRowData {
+}
+
+export interface IExtra extends IExtraData {
+  rowData?: IRowData;
+}
+
+export type IFormatterValueType = formatterValueType & {
+  title?: string | React.ReactNode;
+  props?: any;
+};
+
+export interface ISortBy {
+  index?: number;
+  direction?: 'asc' | 'desc'
+}
+
+export interface IAction extends Omit<DropdownItemProps, 'title' | 'onClick'> {
+  isSeparator?: boolean;
+  itemKey?: string;
+  title?: string | React.ReactNode;
+  onClick: (event: React.MouseEvent, rowIndex: number, rowData: IRowData, extraData: IExtraData) => void;
+}
+
+export interface ISeparator extends IAction {
+  isSeparator: boolean;
+}
+
+export type IActions = (IAction | ISeparator)[];
+export type IActionsResolver = (rowData: IRowData, extraData: IExtraData) => (IAction | ISeparator)[];
+export type IAreActionsDisabled = (rowData: IRowData, extraData: IExtraData) => boolean;
+
+// to be removed in future, this interface is no longer accurate
+export interface IDecorator extends React.HTMLProps<HTMLElement> {	
+  isVisible: boolean;	
+  children?: React.ReactNode;	
+}
+
+export interface ICell {
+  title?: string;
+  transforms?: ((...args: any) => any)[];
+  cellTransforms?: ((...args: any) => any)[];
+  columnTransforms?: ((...args: any) => any)[];
+  formatters?: ((...args: any) => any)[];
+  cellFormatters?: ((...args: any) => any)[];
+  props?: any;
+  data?: any;
+  header?: any;
+  cell?: any;
+}
+
+export interface IRowCell {
+  title?: React.ReactNode;
+  props?: any;
+}
+
+export interface IRow extends RowType {
+  cells?: (React.ReactNode | IRowCell)[];
+  isOpen?: boolean;
+  parent?: number;
+  compoundParent?: number;
+  props?: any;
+  fullWidth?: boolean;
+  noPadding?: boolean;
+  showSelect?: boolean;
+  isExpanded?: boolean;
+  isFirstVisible?: boolean;
+  isLastVisible?: boolean;
+  selected?: boolean;
+}
+
+export interface TableProps {
+  'aria-label'?: string;
+  children?: React.ReactNode;
+  className?: string;
+  variant?: 'compact';
+  borders?: boolean;
+  gridBreakPoint?: '' | 'grid' | 'grid-md' | 'grid-lg' | 'grid-xl' | 'grid-2xl';
+  sortBy?: ISortBy;
+  onCollapse?: OnCollapse;
+  onExpand?: OnExpand;
+  onSelect?: OnSelect;
+  onSort?: OnSort;
+  actions?: IActions;
+  actionResolver?: IActionsResolver;
+  areActionsDisabled?: IAreActionsDisabled;
+  header?: React.ReactNode;
+  caption?: React.ReactNode;
+  rowLabeledBy?: string;
+  expandId?: string;
+  contentId?: string;
+  dropdownPosition?: 'right' | 'left';
+  dropdownDirection?: 'up' | 'down';
+  rows: (IRow | string[])[];
+  cells: (ICell | string)[];
+  bodyWrapper?: Function;
+  rowWrapper?: Function;
+  role?: string;
+}
+
+export const TableContext = React.createContext({
+  headerData: null as any,
```

Comment:
I think this `any` could be `RenderersTypes['columns']` based on the fact that `headerData` is passed down to the `columns` prop of `Provider` which has that type.

Reply:
good one! updating.